distributed/client.py

```diff
@@ -2552,7 +2552,86 @@ def _graph_to_futures(
+
+                    if layer_priority is not False:
+                        if isinstance(layer_priority, Number):
+                            for k in layer:
+                                user_priority[stringify(k)] = layer_priority
+                        elif callable(layer_priority):
+                            for k in layer:
+                                user_priority[stringify(k)] = layer_priority(k)
+                        else:
+                            raise TypeError(
+                                "Expected priority to be a number or callable got %s"
+                                % str(layer_priority)
+                            )
+
+                    layer_workers = layer.annotations.get("workers", False)
+
+                    if layer_workers is not False:
+                        if isinstance(layer_workers, str):
+                            layer_workers = [layer_workers]
+                            for k in layer:
+                                annotation_restrictions[stringify(k)] = layer_workers
+                        elif isinstance(layer_workers, (list, tuple, set)):
+                            layer_workers = list(layer_workers)
+                            for k in layer:
+                                annotation_restrictions[stringify(k)] = layer_workers
+                        else:
+                            raise TypeError(
+                                "Expected workers to be a string or list got %s"
+                                % str(layer_workers)
+                            )
+
+                    layer_loose_restrictions = layer.annotations.get(
+                        "allow_other_workers", False
+                    )
+
+                    if layer_loose_restrictions is True:
+                        annotation_loose_restrictions.extend(
+                            stringify(k) for k in layer
+                        )
+
+                    layer_retries = layer.annotations.get("retries", False)
+
+                    if layer_retries is not False:
+                        if isinstance(layer_retries, Number):
+                            for k in layer:
+                                annotation_retries[stringify(k)] = layer_retries
+                        else:
+                            raise TypeError(
+                                "Expected retries to be a number got %s"
+                                % str(layer_retries)
+                            )
+
+                    layer_resources = layer.annotations.get("resources", False)
+
+                    if layer_resources:
+                        if isinstance(layer_resources, dict):
+                            for k in layer:
+                                annotation_resources[stringify(k)] = layer_resources
+                        else:
+                            raise TypeError("Expected resources to be a dict")
```

Comment:
If we're going to send along the annotations to the scheduler anyway then we might consider doing this work on the scheduler side.

Reply:
I've removed the layer extraction code from ``Client._graph_to_futures`` and reintroduced it into ``Scheduler.update_graph``